pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/creator/SegmentGeneratorConfig.java

```diff
@@ -261,6 +265,17 @@ public static Schema updateSchemaWithTimestampIndexes(Schema schema,
     return newSchema;
   }
 
+  public static Schema updateSchemaWithDerivedFlags(Schema schema) {
```

Comment:
Don't like this method name. Suggestions welcome on renaming it (or if it is needed at all)